_posts/2016-02-25-usage-tests.md

```diff
@@ -0,0 +1,73 @@
+---
+layout: post
+title: Usage Tests
+---
+
+Tests suites have a number of purposes.
+Perhaps most commonly you think of the test suite as validating the behavior of your code.
+Absolutely.
+But they also serve as examples of using the lib.
+Let's make that more explicit.
+
+## Usage
+
+If you have perused the test suites for library code, you may have noticed some conventions.
+Folks tend to have a `units/` directory for unit (read: isolated) tests.
+Other places you may see a `lib/` directory if the author is trying to keep the file to file organization the same between their implementation and tests.
+Finally you may have seen an `integration/`, `acceptance/`, etc. directory that indicates tests of various pieces working together.
+Sometimes the integration directory (whatever it be called) also has tests illustrating example usage of the lib.
+
+Here's the thing.
+Probably all non-trivial, well organized code will have interfaces that aren't meant to be used publicly.
+Within a class we often mark methods as `private` to discourage external use.
+You may even do the same thing [with internal classes][private-class].
+
+Hard as you try, you will always have publicly _accessible_ interfaces that were never intended for external use.
+Recently, I was [working on ActionCable][ac-pr] and a [suggestion] was made that we move the change closer to the library's user interface.
+It became clear with that suggestion how difficult it is to discern what portion of ActionCable was intended to be used directly.
+It's actually quite small.
+
+So how do you communicate this to the reader.
+Perhaps by organizing your test suite better.
+Rather than having to dig through a ton of test files in numerous directories, make it explicit.
+
+## usage_test.(rb|exs|js|java|etc)
+
+Call it like it is.
+If you're testing the usage of you library, it's a usage test.
+You could call it whatever you want, examples, acceptance, etc.
+The point is to communicate the intend, and to be honest if a convention is established that's recognizable everyone wins.
```

Comment:
I know you haven't edited this yet, but just want to make sure you don't miss this s/intend/intent/